### PR TITLE
Support Commonmarker v1 api

### PR DESCRIPTION
**Pull request by unasuke**
## What
retry of https://github.com/rtomayko/tilt/pull/396

> The Commonmarker gem released v1 incluging breaking api changes.
> https://github.com/gjtorikian/commonmarker/releases/tag/v1.0.0
>
> These changes supports the new commonmarker's api.
>
> To support commonmaker's breaking api changes, tilt gem will release new version including incompatible api change also.
> For notice to tilt gem user, is it better to check commonmarker gem's version?

... and support both of `CommonMaker` (0.x) and `Commonmarker` (1.0.x)

### CODE REVIEW

File: .ci.gemfile
Comment (jeremyevans) [inline .ci.gemfile:78]: Is this necessary?  `commonmarker` uses `required_ruby_version`, so Ruby versions < 3.1 should automatically use an older version.  If you really need to force a version, something like:

```
if RUBY_VERSION >= '3.1'
  gem 'commonmarker'
else
  gem 'commonmarker', '< 1'
end
```

should work.

File: .github/workflows/ci.yml
Comment (jeremyevans) [inline .github/workflows/ci.yml:20]: With the `.ci.gemfile` changes recommended, all changes in this file can be removed.

These changes double the number of CI jobs, which I don't want to do.  It's fine to test commonmarker 1.0+ on Ruby 3.1+, and commonmarker <1.0 on Ruby <3.1.

File: test/tilt_commonmarkertemplate_test.rb
Comment (jeremyevans) [inline test/tilt_commonmarkertemplate_test.rb:4]: This should be switched to `if defined?(::Commonmarker)` if you want to test stuff that only works on commonmarker >= 1.  However, in general we would want the same specs to pass in both pre-1.0 and post-1.0 versions of commonmarker.

Adding support for new keys is fine, but can we also map old option keys to new option keys if the new version is used, so the same tilt code can work regardless of the underlying commonmarker version, assuming the new commonmarker supports the same feature as the previous version using a different option name?

File: lib/tilt/commonmarker.rb
Comment (jeremyevans) [inline lib/tilt/commonmarker.rb:38]: We don't want to define different constants for commonmarker >= 1.0.  Ideally, we don't want to define constants at all.  Instead, the local variables previously defined should be changed if commonmarker > 1.0 is used.  You probably want an approach such as:

```ruby
if defined?(::Commonmarker)
  # new setting of parse_opts, render_opts, exts
  # new Tilt::CommonMarkerTemplate = Tilt::StaticTemplate.subclass do
else
  # previous setting of parse_opts, render_opts, exts
  # previous Tilt::CommonMarkerTemplate = Tilt::StaticTemplate.subclass do
end
```

Removing the private methods will also fix the failures in older Ruby versions (tilt supports back to Ruby 2.0).

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for working on this.  This is a great start.  Please see requested changes.  Also, I've enabled CI so you can see where the CI failures are.

File: .ci.gemfile
Comment (unasuke) [reply .ci.gemfile:78]: I thought anyone might want to use less than version 1 of commonmarker with Ruby 3.1 or higher. Is this unnecessary?

Comment (jeremyevans) [reply .ci.gemfile:78]: Certainly users can use commonmarker < 1 on Ruby >= 3.1.  However, we don't need to test that in CI.  I want CI to be reasonably small and fast, with no more than one run per Ruby version.

**Comment by unasuke** [comment]
I made some commits above older changesets for review. I think it's better to squash before it merges.

**Comment by jeremyevans** [comment]
Looks good, thank you for making the changes.  I'll squash merge this assuming CI passes.

**Comment by jeremyevans** [comment]
Looks like a couple spec fixes are needed to pass on Ruby 2.1 and 2.0.  Just switch to a regular heredoc (`<<` instead of `~<<`) and do not indent, that should fix it.

**Comment by unasuke** [comment]
Fixed at 290541d !

**Comment by jeremyevans** [comment]
Thank you very much for your work on this!

**Comment by unasuke** [comment]
@jeremyevans Thank you too, I'm glad if https://github.com/rtomayko/tilt/pull/395 is merged also.

**Comment by jeremyevans** [comment]
The gem homepage was fixed.  I agree that it would be best the old repository README was updated, but I have no control over that I would not expect it at this point.

**Pull request merged**
